qa/tasks/cephfs/volumes/test_clone_stats.py

```diff
@@ -791,3 +791,180 @@ def test_when_clones_cancelled_are_more_than_cloner_threads(self):
                 pass
             else:
                 raise
+
+
+class TestOngoingClonesCounter(CloneProgressReporterHelper):
+    '''
+    Class CloneProgressReporter contains the code that lets it figure out the
+    number of ongoing clones on its own, without referring the MGR config
+    option mgr/volumes/max_concurrenr_clones. This class contains tests to
+    ensure that this code, that does the figuring out, is working fine.
+    '''
+
+    def test_for_2_ongoing_clones(self):
```

Comment:
can't the three test methods:
```
def test_for_2_ongoing_clones()
def test_for_4_ongoing_clones()
def test_for_6_ongoing_clones()
```
be factored out of a helper function with necessary arguments ?

Reply:
Going through all of them again, I think it can be done but I would like to avoid it since the helper method can go through too much branching to cover all three test methods. Also, if a bug is discovered in one test method in future, modifying the helper method to fix the bug can potentially lead to bugs in other two test methods.

Reply:
I agree which is why I think it can be done. But what usually happens is when we modify helper method that contains test code due to bug in 1 test, it undesirably and unnecessarily affects other tests causing developer to now check other tests as well. So I would like to avoid that and keep these methods as it is.

Reply:
I'll still strongly suggest you to factor out the common parts into a helper function and parameterize it to help run the three tests.

Reply:
Okay, done.